Detailed spec: auction & profit distribution

agryaznov opened the pull request:
1. Described auction mechanics in detail

2. Only 1 ticket per bidding ETH address

    if we don't apply this rule, person who made several bids wishing only one ticket will get several tickets reserved to him, dropping out other people with lower bids

3. Added Profit Distribution formalized rules

4. For the sake of simplicity, we do not count costs explicitly in the contract. Costs are Organizer's risk, and are accounted in her revenue share rate.

5. We need to specify checkin process mechanics in detail. @xhipster @dearnadia please go ahead and write your vision of it here 

6. We need to describe process of speaker approval in detail. What are the criterias? How we garantee that speaker's deposit do not affect the decision (taking into account the fact that ecomonically it actually should affect). @xhipster @dearnadia again, please go ahead and write your vision of it here

agryaznov committed:
Detalized spec: auction & profit distribution

mastercyb committed:
my variant of profit distribution formula

mastercyb commented:
@DearNadia Please, review this pl too

agryaznov commented:
@xhipster there's some formatting issues with the formula, making it looks like having no sense... pls fix it

agryaznov commented:
@xhipster @DearNadia any ideas regarding checkin mechanics?

mastercyb commented:
I suggested a process in our chat. Let's discuss first

mastercyb committed:
Fix formatting issues with the formulas

mastercyb commented:
> @xhipster there's some formatting issues with the formula, making it looks like having no sense... pls  fix it

[fixed](https://github.com/cyberacademia/cyberevents/pull/40/commits/2e57a7299b35301416e61e4615463f4bdb68b2be)

agryaznov commented:
> [fixed](https://github.com/cyberacademia/cyberevents/pull/40/commits/2e57a7299b35301416e61e4615463f4bdb68b2be)

@xhipster I'm not sure <speaker's profit> formula is correct there ... there should be some division on number of checked-in speakers, no? I mean formula should give us profit for particular speaker, not for all of them, right?

agryaznov committed:
+ checkin rules&mechanics

agryaznov commented:
+ added checkin rules&mechanics
+ 1 checkin closes 1 hour before \<*timestamp of profit distribution*\>

   * checked-in speakers and attendees are written to contract with corresponding transactions
    * no etry allowed from this moment

+ added auction losing bids return rules
